support/reftest/reftest

```diff
@@ -0,0 +1,249 @@
+    )
+    # parser for prepare operation
+    parser_prepare = subparsers.add_parser(
+        "prepare",
+        help="""
+        prepare reference test data in dynamodb and s3
+        for integration test,
+         e.g. $./reftest prepare --bucket exodus-bucket --table exodus-table
+        """,
+    )
+    parser_prepare.add_argument(
+        "--release-date",
+        default=None,
+        help="Date on which the content will be made available.",
+    )
+    parse_aws_session(parser_prepare)
+    parse_cdn_certs(parser_prepare)
+    parser_prepare.add_argument(
+        "--bucket",
+        required=True,
+        help="The AWS S3 bucket used to store test data",
+    )
+    parser_prepare.add_argument(
+        "--table",
+        required=True,
+        help="The AWS dynamoDB used to store test data",
+    )
+
+    return root_parser.parse_args()
+
+
+class RefTestConfig:
+    def __init__(self, prod_cdn_url, test_data):
+        self.prod_cdn_url = prod_cdn_url
+        self.test_data = test_data
+
+
+def load_config():
+    with open("data.yml") as f:
+        config = yaml.load(f, yaml.SafeLoader)
+    return RefTestConfig(config["prod-cdn-url"], config["test_data"],)
+
+
+def download_to_local(url, key_path, cert_path, cacert_path):
+    with requests.get(
+        url,
+        cert=(cert_path, key_path),
+        verify=cacert_path,
+        stream=True,
+        timeout=(30, 30),
+    ) as r:
+        r.raise_for_status()
+        total_size = int(r.headers.get("content-length", 0))
+        t = tqdm(desc=url, total=total_size, unit="iB", unit_scale=True)
+
+        # the file is deleted as soon as it is closed.
+        temp_file = tempfile.NamedTemporaryFile(delete=True)
+        for chunk in r.iter_content(chunk_size=8192):
+            if chunk:  # filter out keep-alive new chunks
+                t.update(len(chunk))
+                temp_file.write(chunk)
+                temp_file.flush()
+        t.close()
+        return temp_file
+
+
+# verify cdn data checksum mathch test data checksum
+def verify_sha256_checksum(temp_file_path, test_data_checksum):
+    if os.path.isfile(temp_file_path):
+        sha256 = hashlib.sha256()
+        with open(temp_file_path, "rb") as binary:
+            while True:
+                chunk = binary.read(4096)
+                if not chunk:
+                    break
+                sha256.update(chunk)
+
+        cdn_data_checksum = sha256.hexdigest()
+
+        if cdn_data_checksum == test_data_checksum:
+            return True
+
+    return False
+
+
+def prepare(db, s3, config, opt):
+    for item in config.test_data:
+        url = config.prod_cdn_url + item["path"]
+        # download test data to local with a name "NamedTemporaryFile"
+        temp_file = download_to_local(url, opt.key, opt.cert, opt.cacert)
+
+        # checksum verify
+        if not verify_sha256_checksum(temp_file.name, item["sha256"]):
+            print("{} verify checksum failed".format(item["path"]))
```

Comment:
I think you should at least log the actual checksum here.  If you don't then it seems like the problem can't be investigated further afterward, and you don't have the checksum you need in order to update the reference test data (because the temp file was already deleted by the time the script exits...)